<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sabrina-mi" />

<meta name="date" content="2023-04-04" />

<title>Br EN model Validation</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PTRS-Rat-Analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sabrina-mi/PTRS-Rat-Analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Br EN model Validation</h1>
<h4 class="author">sabrina-mi</h4>
<h4 class="date">2023-04-04</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-04-04
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>PTRS-Rat-Analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity itâ€™s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230404code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230404)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230404code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230404)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsabrinamiPTRSRatAnalysistree7dd0181ef955b659eb123f567613a6dc1c1ff8c1targetblank7dd0181a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sabrina-mi/PTRS-Rat-Analysis/tree/7dd0181ef955b659eb123f567613a6dc1c1ff8c1" target="_blank">7dd0181</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsabrinamiPTRSRatAnalysistree7dd0181ef955b659eb123f567613a6dc1c1ff8c1targetblank7dd0181a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sabrina-mi/PTRS-Rat-Analysis/tree/7dd0181ef955b659eb123f567613a6dc1c1ff8c1" target="_blank">7dd0181</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store

</code></pre>
<p>
Note that any generated files, e.g.Â HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Br_EN_Validation.Rmd</code>) and
HTML (<code>docs/Br_EN_Validation.html</code>) files. If youâ€™ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sabrina-mi/PTRS-Rat-Analysis/blob/7dd0181ef955b659eb123f567613a6dc1c1ff8c1/analysis/Br_EN_Validation.Rmd" target="_blank">7dd0181</a>
</td>
<td>
sabrina-mi
</td>
<td>
2023-04-04
</td>
<td>
migrated box analysis files
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(readr)
library(tidyverse)</code></pre>
<pre><code>â”€â”€ Attaching core tidyverse packages â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€ tidyverse 2.0.0 â”€â”€
âœ” dplyr     1.1.1     âœ” purrr     1.0.1
âœ” forcats   1.0.0     âœ” stringr   1.5.0
âœ” ggplot2   3.4.1     âœ” tibble    3.2.1
âœ” lubridate 1.9.2     âœ” tidyr     1.3.0
â”€â”€ Conflicts â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€ tidyverse_conflicts() â”€â”€
âœ– dplyr::filter() masks stats::filter()
âœ– dplyr::lag()    masks stats::lag()
â„¹ Use the ]8;;http://conflicted.r-lib.org/conflicted package]8;; to force all conflicts to become errors</code></pre>
<pre class="r"><code>&quot;%&amp;%&quot; = function(a,b) paste(a,b,sep=&quot;&quot;)
dir=&quot;/Users/sabrinami/Library/CloudStorage/Box-Box/rat-genomic-analysis/PTRS-2023/RatXcan/&quot;</code></pre>
<pre class="r"><code>library(stringr)
filelist &lt;- list.files(dir %&amp;% &quot;Results/associations&quot;, pattern = &quot;Br__association_&quot;, full.names = TRUE)
full_df &lt;- data.frame()
for(file in filelist) {
  assoc_file &lt;- suppressMessages(read_tsv(file, col_names = TRUE))
  # extract phenotype from regex matching in file name
  pheno_id &lt;- str_match(file, &quot;association_(.*?).txt&quot;)[,2]
  tempo &lt;- cbind(assoc_file, metabolic_trait=pheno_id) %&gt;% select(-c(status))
  full_df &lt;- rbind(full_df, tempo)
} 

saveRDS(full_df, dir %&amp;% &quot;assoc_full_df.RDS&quot;)</code></pre>
<div id="significant-genes" class="section level2">
<h2>Significant genes</h2>
<p>number of significant associations for each gene</p>
<pre class="r"><code>tempo_df &lt;- full_df %&gt;% filter(pvalue &lt; .05/5601)


#428 sig genes
tempo_df %&gt;% group_by(gene) %&gt;% summarise(n = n())</code></pre>
<pre><code># A tibble: 32 Ã— 2
   gene                   n
   &lt;chr&gt;              &lt;int&gt;
 1 ENSRNOG00000000511     1
 2 ENSRNOG00000000795     1
 3 ENSRNOG00000000816     1
 4 ENSRNOG00000000831     1
 5 ENSRNOG00000000975     2
 6 ENSRNOG00000000977     1
 7 ENSRNOG00000001047     2
 8 ENSRNOG00000001069     1
 9 ENSRNOG00000001159     2
10 ENSRNOG00000001194     2
# â„¹ 22 more rows</code></pre>
<p>number significant genes for each phenotype</p>
<pre class="r"><code>tempo_df %&gt;% group_by(metabolic_trait) %&gt;% summarise(n = n())</code></pre>
<pre><code># A tibble: 2 Ã— 2
  metabolic_trait       n
  &lt;chr&gt;             &lt;int&gt;
1 length_w_tail_cm     26
2 length_wo_tail_cm    11</code></pre>
</div>
<div id="distribution-of-prediction-performances"
class="section level2">
<h2>distribution of prediction performances</h2>
<pre class="r"><code>suppressMessages(library(tidyverse))
suppressMessages(library(glue))
PRE = &quot;/Users/sabrinami/Library/CloudStorage/Box-Box/rat-genomic-analysis/PTRS-2023&quot;


library(&quot;RSQLite&quot;)
sqlite &lt;- dbDriver(&quot;SQLite&quot;)
dbname &lt;- glue(&quot;{PRE}/RatXcan/Br_elasticNet.db&quot;) ## add full path if db file not in current directory
## connect to db
db = dbConnect(sqlite,dbname)
## list tables
dbListTables(db)</code></pre>
<pre><code>[1] &quot;construction&quot; &quot;extra&quot;        &quot;sample_info&quot;  &quot;weights&quot;     </code></pre>
<pre class="r"><code>## convenience query function
query &lt;- function(...) dbGetQuery(db, ...)
## example queries
query(&#39;select count(*) from weights&#39;)</code></pre>
<pre><code>  count(*)
1    14224</code></pre>
<p>14,224 snp-gene pairs.</p>
<pre class="r"><code>weights = dbGetQuery(db,&quot;select * from weights&quot;)
glimpse(weights)</code></pre>
<pre><code>Rows: 14,224
Columns: 6
$ gene       &lt;chr&gt; &quot;ENSRNOG00000040300&quot;, &quot;ENSRNOG00000040300&quot;, &quot;ENSRNOG0000005â€¦
$ rsid       &lt;chr&gt; &quot;1_1643610_C_T&quot;, &quot;1_2092364_C_T&quot;, &quot;1_1643610_C_T&quot;, &quot;1_20923â€¦
$ varID      &lt;chr&gt; &quot;1_1643610_C_T&quot;, &quot;1_2092364_C_T&quot;, &quot;1_1643610_C_T&quot;, &quot;1_20923â€¦
$ ref_allele &lt;chr&gt; &quot;C&quot;, &quot;C&quot;, &quot;C&quot;, &quot;C&quot;, &quot;C&quot;, &quot;C&quot;, &quot;C&quot;, &quot;C&quot;, &quot;C&quot;, &quot;C&quot;, &quot;C&quot;, &quot;C&quot;,â€¦
$ eff_allele &lt;chr&gt; &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;,â€¦
$ weight     &lt;dbl&gt; -0.199572530, 0.098030720, -0.123155000, 0.238959200, -0.25â€¦</code></pre>
<pre class="r"><code>extra = dbGetQuery(db,&quot;select * from extra&quot;)
glimpse(extra)</code></pre>
<pre><code>Rows: 5,601
Columns: 6
$ gene            &lt;chr&gt; &quot;ENSRNOG00000040300&quot;, &quot;ENSRNOG00000050129&quot;, &quot;ENSRNOG00â€¦
$ genename        &lt;chr&gt; &quot;Raet1e&quot;, &quot;AABR07000137.1&quot;, &quot;AABR07000145.1&quot;, &quot;AABR070â€¦
$ n.snps.in.model &lt;int&gt; 2, 2, 2, 1, 2, 1, 2, 2, 2, 2, 2, 1, 2, 2, 2, 2, 1, 2, â€¦
$ pred.perf.R2    &lt;dbl&gt; 2.687766e-02, 4.222472e-02, 3.455208e-02, 1.769772e-03â€¦
$ pred.perf.pval  &lt;dbl&gt; 1.339141e-02, 1.857276e-03, 4.959378e-03, 5.282966e-01â€¦
$ pred.perf.qval  &lt;int&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NAâ€¦</code></pre>
<pre class="r"><code>construction = dbGetQuery(db,&quot;select * from construction &quot;)
sample_info = dbGetQuery(db,&quot;select * from sample_info&quot;)
nsam = sample_info$n.samples

weights %&gt;% count(gene) %&gt;% arrange(desc(n)) %&gt;% head(20)</code></pre>
<pre><code>                 gene  n
1  ENSRNOG00000047706 22
2  ENSRNOG00000000451 18
3  ENSRNOG00000000436 16
4  ENSRNOG00000026799 16
5  ENSRNOG00000000478 15
6  ENSRNOG00000000433 14
7  ENSRNOG00000000841 14
8  ENSRNOG00000022283 14
9  ENSRNOG00000027157 14
10 ENSRNOG00000061639 14
11 ENSRNOG00000000432 13
12 ENSRNOG00000054031 13
13 ENSRNOG00000058784 13
14 ENSRNOG00000059770 13
15 ENSRNOG00000029500 12
16 ENSRNOG00000039744 12
17 ENSRNOG00000056637 12
18 ENSRNOG00000062067 12
19 ENSRNOG00000000423 11
20 ENSRNOG00000000435 11</code></pre>
<pre class="r"><code>weights %&gt;% count(gene) %&gt;% arrange(desc(n)) %&gt;% .[[&quot;n&quot;]] %&gt;% summary()</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   1.00    2.00    2.00    2.54    3.00   22.00 </code></pre>
</div>
<div id="performance-of-predictors" class="section level2">
<h2>performance of predictors</h2>
<pre class="r"><code>devtools::source_gist(&quot;38431b74c6c0bf90c12f&quot;)</code></pre>
<pre><code>â„¹ Sourcing gist &quot;38431b74c6c0bf90c12f&quot;
â„¹ SHA-1 hash of file is &quot;cbeca7fd9bf1602dee41c4f1880cc3a5e8992303&quot;</code></pre>
<pre class="r"><code>qqunif(extra$pred.perf.pval)</code></pre>
<pre><code>Warning in qqunif(extra$pred.perf.pval): thresholding p to 1e-30</code></pre>
<p><img src="figure/Br_EN_Validation.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>devtools::source_gist(&quot;a925fea01b365a8c605e&quot;)</code></pre>
<pre><code>â„¹ Sourcing gist &quot;a925fea01b365a8c605e&quot;
â„¹ SHA-1 hash of file is &quot;01da6a099ce8e3272b3036fcf075b7afa5287dad&quot;</code></pre>
<pre class="r"><code>qqR2(extra$pred.perf.R2,nsam,main=&quot;R2 vs expected&quot;)</code></pre>
<p><img src="figure/Br_EN_Validation.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="well-predicted-genes" class="section level2">
<h2>well-predicted genes</h2>
<p>Using qvalue &lt; .05</p>
<pre class="r"><code>library(qvalue)
qq = qvalue(extra$pred.perf.pval)
#hist(qq$qvalues)</code></pre>
<pre class="r"><code>extra$pred.perf.qval = qq$qvalues
high_qual_genes = filter(extra, extra$pred.perf.qval &lt; .05)
hist(high_qual_genes$pred.perf.R2, main=&quot;Histogram of R2 for qval &lt; genes&quot;)</code></pre>
<p><img src="figure/Br_EN_Validation.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparing-well-predicted-genes" class="section level2">
<h2>Comparing well-predicted genes</h2>
<p>R2 generated from initial analysis</p>
<pre class="r"><code>library(readxl)
R2 = suppressWarnings(read_excel(&quot;/Users/sabrinami/Library/CloudStorage/Box-Box/rat-genomic-analysis/PTRS-2023/RatXcan/Results/santhanam-et-al-2022_supplementary-tables-ssr.xlsx&quot;, sheet = &quot;ST15 R2 betweeen Tissues&quot;, skip=1))

hist(R2$`R2 in NAcc`, main=&quot;Histogram of R2 NAcc&quot;)</code></pre>
<p><img src="figure/Br_EN_Validation.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>full_R2 = inner_join(extra, R2, by=c(&quot;gene&quot;=&quot;Rat Ensembl ID&quot;))
full_R2 = select(full_R2, c(&quot;gene&quot;, &quot;pred.perf.R2&quot;, &quot;R2 in NAcc&quot;, &quot;R2 in IL&quot;, &quot;R2 in LHb&quot;, &quot;R2 in PL&quot;, &quot;R2 in OFC&quot;))</code></pre>
</div>
<div id="predict.py-performance" class="section level1">
<h1>Predict.py Performance</h1>
<p>(The results here are a good sign for the model, but statistically
meaningless, because the testing set is the same as the training
set)</p>
<pre class="r"><code>pred_expr = read.table(&quot;/Users/sabrinami/Library/CloudStorage/Box-Box/rat-genomic-analysis/PTRS-2023/RatXcan/Results/Br_EN__predict.txt&quot;, header=TRUE)
obs_expr = read.csv(&quot;/Users/sabrinami/Library/CloudStorage/Box-Box/rat-genomic-analysis/PTRS-2023/RatData/phenotypes/gexBr.csv&quot;,header = TRUE)

gene_list = colnames(pred_expr)[-c(1,2)]
df = data.frame(gene = gene_list, cor= vector(&quot;numeric&quot;, 5601))
for(i in 3:5603){
  gene=colnames(pred_expr)[i]
  pred = pred_expr[,c(1,i)]
  obs = obs_expr[,c(&quot;X&quot;,gene)]
  temp = inner_join(pred,obs, by=c(&quot;FID&quot; = &quot;X&quot;))
  colnames(temp) = c(&quot;FID&quot;, &quot;Predicted&quot;, &quot;Observed&quot;)
  r = cor(temp$Predicted,temp$Observed)
  df[i-2,2]=r
}

hist(df$cor)</code></pre>
<pre class="r"><code>devtools::source_gist(&quot;a925fea01b365a8c605e&quot;)
qqR(df$cor,nsam,main=&quot;correlation vs expected&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 (2022-10-31)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] readxl_1.4.2    qvalue_2.30.0   RSQLite_2.3.0   glue_1.6.2     
 [5] lubridate_1.9.2 forcats_1.0.0   stringr_1.5.0   dplyr_1.1.1    
 [9] purrr_1.0.1     tidyr_1.3.0     tibble_3.2.1    ggplot2_3.4.1  
[13] tidyverse_2.0.0 readr_2.1.4     workflowr_1.7.0

loaded via a namespace (and not attached):
 [1] fs_1.6.1          usethis_2.1.6     devtools_2.4.5    bit64_4.0.5      
 [5] httr_1.4.5        rprojroot_2.0.3   gh_1.4.0          tools_4.2.2      
 [9] profvis_0.3.7     bslib_0.4.2       utf8_1.2.3        R6_2.5.1         
[13] DBI_1.1.3         colorspace_2.1-0  urlchecker_1.0.1  withr_2.5.0      
[17] tidyselect_1.2.0  prettyunits_1.1.1 processx_3.8.0    bit_4.0.5        
[21] curl_5.0.0        compiler_4.2.2    git2r_0.31.0      httr2_0.2.2      
[25] cli_3.6.0         sass_0.4.5        scales_1.2.1      callr_3.7.3      
[29] rappdirs_0.3.3    digest_0.6.31     rmarkdown_2.20    pkgconfig_2.0.3  
[33] htmltools_0.5.4   sessioninfo_1.2.2 fastmap_1.1.1     highr_0.10       
[37] htmlwidgets_1.6.2 rlang_1.1.0       rstudioapi_0.14   shiny_1.7.4      
[41] jquerylib_0.1.4   generics_0.1.3    jsonlite_1.8.4    vroom_1.6.1      
[45] magrittr_2.0.3    Rcpp_1.0.10       munsell_0.5.0     fansi_1.0.4      
[49] lifecycle_1.0.3   stringi_1.7.12    whisker_0.4.1     yaml_2.3.7       
[53] pkgbuild_1.4.0    plyr_1.8.8        grid_4.2.2        blob_1.2.4       
[57] parallel_4.2.2    promises_1.2.0.1  crayon_1.5.2      miniUI_0.1.1.1   
[61] splines_4.2.2     hms_1.1.3         knitr_1.42        ps_1.7.3         
[65] pillar_1.9.0      reshape2_1.4.4    pkgload_1.3.2     evaluate_0.20    
[69] getPass_0.2-2     remotes_2.4.2     vctrs_0.6.1       tzdb_0.3.0       
[73] httpuv_1.6.9      cellranger_1.1.0  gtable_0.3.3      cachem_1.0.7     
[77] xfun_0.37         mime_0.12         xtable_1.8-4      gitcreds_0.1.2   
[81] later_1.3.0       memoise_2.0.1     timechange_0.2.0  ellipsis_0.3.2   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
