<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sabrina-mi" />

<meta name="date" content="2023-04-04" />

<title>Predict Rat Traits with PTRS</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PTRS-Rat-Analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sabrina-mi/PTRS-Rat-Analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Predict Rat Traits with PTRS</h1>
<h4 class="author">sabrina-mi</h4>
<h4 class="date">2023-04-04</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-04-24
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>PTRS-Rat-Analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230404code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230404)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230404code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230404)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsabrinamiPTRSRatAnalysistree808f440df0928f3f3faad096810505c90f8fe06ftargetblank808f440a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sabrina-mi/PTRS-Rat-Analysis/tree/808f440df0928f3f3faad096810505c90f8fe06f" target="_blank">808f440</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsabrinamiPTRSRatAnalysistree808f440df0928f3f3faad096810505c90f8fe06ftargetblank808f440a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sabrina-mi/PTRS-Rat-Analysis/tree/808f440df0928f3f3faad096810505c90f8fe06f" target="_blank">808f440</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    GEMMA/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    code/.DS_Store

Untracked files:
    Untracked:  code/Ac_GEMMA_BSLMM.sh
    Untracked:  code/generate_GRMs.yaml
    Untracked:  code/run_BSLMM.yaml
    Untracked:  code/subset_geno.awk
    Untracked:  code/write_GEMMA_inputs.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/PTRS_Predict.Rmd</code>) and HTML
(<code>docs/PTRS_Predict.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sabrina-mi/PTRS-Rat-Analysis/808f440df0928f3f3faad096810505c90f8fe06f/docs/PTRS_Predict.html" target="_blank">808f440</a>
</td>
<td>
sabrina-mi
</td>
<td>
2023-04-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sabrina-mi/PTRS-Rat-Analysis/blob/d5812dc360bbd2723c21a0067eb4f267f072cb6c/analysis/PTRS_Predict.Rmd" target="_blank">d5812dc</a>
</td>
<td>
sabrina-mi
</td>
<td>
2023-04-24
</td>
<td>
wflow_publish("analysis/PTRS_Predict.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sabrina-mi/PTRS-Rat-Analysis/67880745b718750d4021b23b5e1f884a61609ca0/docs/PTRS_Predict.html" target="_blank">6788074</a>
</td>
<td>
sabrina-mi
</td>
<td>
2023-04-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sabrina-mi/PTRS-Rat-Analysis/791b48ea19824e790d7043c37e2949c951a4fcae/docs/PTRS_Predict.html" target="_blank">791b48e</a>
</td>
<td>
sabrina-mi
</td>
<td>
2023-04-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sabrina-mi/PTRS-Rat-Analysis/blob/4ff051d477196339bedc18be561ddad1d785c075/analysis/PTRS_Predict.Rmd" target="_blank">4ff051d</a>
</td>
<td>
sabrina-mi
</td>
<td>
2023-04-24
</td>
<td>
wflow_publish("analysis/PTRS_Predict.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sabrina-mi/PTRS-Rat-Analysis/5db94f805593bca294b6e39fb4688cfc95dbd017/docs/PTRS_Predict.html" target="_blank">5db94f8</a>
</td>
<td>
sabrina-mi
</td>
<td>
2023-04-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sabrina-mi/PTRS-Rat-Analysis/blob/7dd0181ef955b659eb123f567613a6dc1c1ff8c1/analysis/PTRS_Predict.Rmd" target="_blank">7dd0181</a>
</td>
<td>
sabrina-mi
</td>
<td>
2023-04-04
</td>
<td>
migrated box analysis files
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.1     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.1     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.1     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the ]8;;http://conflicted.r-lib.org/conflicted package]8;; to force all conflicts to become errors</code></pre>
<pre class="r"><code>library(data.table)</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;

The following objects are masked from &#39;package:lubridate&#39;:

    hour, isoweek, mday, minute, month, quarter, second, wday, week,
    yday, year

The following objects are masked from &#39;package:dplyr&#39;:

    between, first, last

The following object is masked from &#39;package:purrr&#39;:

    transpose</code></pre>
<pre class="r"><code>library(RSQLite)
library(glmnet)</code></pre>
<pre><code>Loading required package: Matrix

Attaching package: &#39;Matrix&#39;

The following objects are masked from &#39;package:tidyr&#39;:

    expand, pack, unpack

Loaded glmnet 4.1-6</code></pre>
<pre class="r"><code>library(data.table)
&quot;%&amp;%&quot; = function(a,b) paste(a,b,sep=&quot;&quot;)
devtools::source_gist(&quot;ee5f67abddd0b761ee24410ea71c41aa&quot;)</code></pre>
<pre><code>ℹ Sourcing gist &quot;ee5f67abddd0b761ee24410ea71c41aa&quot;
ℹ SHA-1 hash of file is &quot;c56941964697a118c351c3f81322a221fa13a1b3&quot;</code></pre>
<pre class="r"><code>devtools::source_gist(&quot;38431b74c6c0bf90c12f&quot;)</code></pre>
<pre><code>ℹ Sourcing gist &quot;38431b74c6c0bf90c12f&quot;
ℹ SHA-1 hash of file is &quot;cbeca7fd9bf1602dee41c4f1880cc3a5e8992303&quot;</code></pre>
<pre class="r"><code>devtools::source_gist(&quot;1e9053c8f35c30396429350a08f33ea7&quot;)</code></pre>
<pre><code>ℹ Sourcing gist &quot;1e9053c8f35c30396429350a08f33ea7&quot;
ℹ SHA-1 hash of file is &quot;7388784ab8c7c2dc5c3f950dc8a47a1c76e3d7ac&quot;</code></pre>
<div id="predict-traits-with-yanyus-ptrs-models" class="section level1">
<h1>Predict Traits with Yanyu’s PTRS models</h1>
<pre class="r"><code>traits &lt;- c(&quot;height&quot;, &quot;bmi&quot;)
dir=&quot;/Users/sabrinami/Library/CloudStorage/Box-Box/rat-genomic-analysis/PTRS-2023/&quot;

# folder with PrediXcan results
results.dir &lt;- dir %&amp;% &quot;RatXcan/Results/&quot;
# folder with PTRS weights, predicted traits will output here
data.dir &lt;- dir %&amp;%&quot;RatData/&quot;</code></pre>
<div id="data-wrangling" class="section level2">
<h2>Data Wrangling</h2>
<pre class="r"><code>orth.rats &lt;- suppressMessages(read_tsv(dir %&amp;% &quot;RatXcan/ortholog_genes_rats_humans.tsv&quot;))

pred_expr &lt;- suppressMessages(read_tsv(results.dir %&amp;% &quot;Br_EN__predict.txt&quot;))

pred_expr$IID = pred_expr$FID

#filter only for genes that have a human ortholog
pred_expr &lt;- pred_expr %&gt;% select(c(IID, intersect(colnames(pred_expr), orth.rats$rnorvegicus_homolog_ensembl_gene) ))


## workaround, since pred_expr

#change name to human ensembl id in humans
colnames(pred_expr)[2:ncol(pred_expr)] &lt;- orth.rats[match(colnames(pred_expr)[2:ncol(pred_expr)], orth.rats$rnorvegicus_homolog_ensembl_gene), 1] %&gt;% .[[&quot;ensembl_gene_id&quot;]]</code></pre>
<pre class="r"><code>fn_weights = function(trait)
{
  weights &lt;- read_tsv(dir %&amp;% &quot;PTRS-models/elastic_net_alpha_0.1_British.export_model/weights.&quot; %&amp;% trait %&amp;% &quot;.tsv.gz&quot;)
  weights$gene_id &lt;- sapply(strsplit(weights$gene_id, &quot;\\.&quot;), `[`, 1)
  rownames(weights) &lt;- weights$gene_id
  weights &lt;- weights %&gt;% rename(gene_name = gene_id)
  return(weights)
}</code></pre>
<pre class="r"><code>for(trait in traits) {
  weights &lt;- fn_weights(trait)
  pred_trait &lt;- fn_generate_trait(pred_expr, weights)
  saveRDS(pred_trait, results.dir %&amp;% &quot;rat_pred_&quot; %&amp;% trait %&amp;% &quot;_w_Human_best_PTRS.RDS&quot;)
  return(pred_trait)
}</code></pre>
</div>
<div id="compute-correlations" class="section level2">
<h2>compute correlations</h2>
<pre class="r"><code>fn_corr = function(pheno, pred_trait, n_genes, RDS){
  trait_df &lt;- data.frame(estimate = numeric(), pvalue = numeric(), model = character(), n_genes_in_model = numeric(), conf.int.min = numeric(), conf.int.max = numeric())
  tempo &lt;- pred_trait[na.omit(match(pheno[,1], rownames(pred_trait))), ]
  for(i in 1:ncol(tempo)){
    cor = cor.test(pheno[,2], tempo[,i])
    trait_df[i,1] &lt;- cor$estimate
    trait_df[i,2] &lt;- cor$p.value
    trait_df[i,3] &lt;- paste(&quot;model&quot;, i, sep = &quot;_&quot;)
    trait_df[i,4] &lt;- n_genes[i,1]
    trait_df[i,5] &lt;- cor$conf.int[1]
    trait_df[i,6] &lt;- cor$conf.int[2]
    saveRDS(trait_df, RDS)
  }
  return(trait_df)
} </code></pre>
<pre class="r"><code>fn_corr_plot = function(file,title){
  trait_df &lt;- readRDS(file)
  p1 = ggplot(trait_df, aes(n_genes_in_model, estimate)) + geom_errorbar(aes(ymin = conf.int.min, ymax = conf.int.max ), width=0.2,  color=&quot;gray&quot;) + geom_point(color = &quot;purple&quot;, position=&quot;jitter&quot;) + geom_line(color = &quot;purple&quot;)   + xlab(&quot;Number of genes in each model&quot;) + ylab(&quot;Correlation Coefficient (r)&quot;) + ggtitle(title) + theme_bw()
  return(p1)
}</code></pre>
</div>
<div id="plot-ptrs-correlations-for-rats" class="section level2">
<h2>Plot PTRS Correlations for Rats</h2>
<pre class="r"><code>pheno_file &lt;- read.csv(data.dir %&amp;% &quot;phenotypes/obesity_upublished_data_raw_phenotypes_PalmerLab.csv&quot;)</code></pre>
<pre class="r"><code>traits &lt;- c(&quot;height&quot;)
rat_analogue &lt;- c(&quot;length_wo_tail_cm&quot;)
names(rat_analogue) &lt;- traits</code></pre>
<pre class="r"><code>for (trait in traits) {
  # data wrangling
  pheno &lt;- pheno_file %&gt;% dplyr::select(c(rfid, rat_analogue[[trait]])) %&gt;% na.omit()
  weights &lt;- read_tsv(dir %&amp;% &quot;PTRS-models/elastic_net_alpha_0.1_British.export_model/weights.&quot; %&amp;% trait %&amp;% &quot;.tsv.gz&quot;) %&gt;% rename(gene_name = gene_id)
  n_genes = as.matrix(apply(weights[,2:ncol(weights)], 2, function(x) sum(x != 0 )))
  # corr inputs
  n_genes &lt;- n_genes[-1, , drop = FALSE]
  pred_trait &lt;- readRDS(results.dir %&amp;% &quot;rat_pred_&quot; %&amp;% trait %&amp;% &quot;_w_Human_best_PTRS.RDS&quot;)[,-1]
  pheno &lt;- pheno[na.omit(match(rownames(pred_trait), pheno$rfid)),]
  RDS &lt;- results.dir %&amp;% &quot;corr_&quot; %&amp;% rat_analogue[[trait]] %&amp;% &quot;_rat_PTRS.RDS&quot;
  # output dataframe for plot
  trait_df &lt;- fn_corr(pheno, pred_trait, n_genes, RDS)
  # plot
}
pred_length_w_tail=readRDS(results.dir %&amp;% &quot;rat_pred_&quot; %&amp;% trait %&amp;% &quot;_w_Human_best_PTRS.RDS&quot;)</code></pre>
<pre class="r"><code>fn_corr_plot(results.dir %&amp;% &quot;corr_length_w_tail_cm_rat_PTRS.RDS&quot;, &quot;Performance of PTRS for Rat Bodylength with Tail&quot;)</code></pre>
<pre><code>Warning: Removed 1 rows containing missing values (`geom_point()`).</code></pre>
<pre><code>Warning: Removed 1 row containing missing values (`geom_line()`).</code></pre>
<p><img src="figure/PTRS_Predict.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sabrina-mi/PTRS-Rat-Analysis/blob/791b48ea19824e790d7043c37e2949c951a4fcae/docs/figure/PTRS_Predict.Rmd/unnamed-chunk-7-1.png" target="_blank">791b48e</a>
</td>
<td>
sabrina-mi
</td>
<td>
2023-04-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>fn_corr_plot(results.dir %&amp;% &quot;corr_length_wo_tail_cm_rat_PTRS.RDS&quot;, &quot;Performance of PTRS for Rat Bodylength without Tail&quot;)</code></pre>
<pre><code>Warning: Removed 1 rows containing missing values (`geom_point()`).</code></pre>
<pre><code>Warning: Removed 1 row containing missing values (`geom_line()`).</code></pre>
<p><img src="figure/PTRS_Predict.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sabrina-mi/PTRS-Rat-Analysis/blob/791b48ea19824e790d7043c37e2949c951a4fcae/docs/figure/PTRS_Predict.Rmd/unnamed-chunk-8-1.png" target="_blank">791b48e</a>
</td>
<td>
sabrina-mi
</td>
<td>
2023-04-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="festus-ptrs" class="section level1">
<h1>Festus PTRS</h1>
<pre class="r"><code>library(rhdf5)
library(data.table)
library(tidyverse)
res.dir = &quot;/Users/sabrinami/Library/CloudStorage/Box-Box/rat-genomic-analysis/PTRS-2023/PTRS-models&quot;
in.file &lt;- glue::glue(&quot;{res.dir}/spxcan2ptrs_original_scale_pt.GIANT_HEIGHT_YENGO_2022.Muscle_Skeletal.results.h5&quot;)
h5ls(in.file)</code></pre>
<pre><code>       group         name       otype dclass       dim
0          /    dataset_0   H5I_GROUP                 
1 /dataset_0      betahat H5I_DATASET  FLOAT 11 x 7662
2 /dataset_0 pval_cutoffs H5I_DATASET  FLOAT        11
3          /        genes H5I_DATASET STRING      7662</code></pre>
<pre class="r"><code>h5.data &lt;- h5read(in.file, &quot;/&quot;)
bb = h5.data[[&quot;dataset_0&quot;]][[&quot;betahat&quot;]]
ll = h5.data[[&quot;dataset_0&quot;]][[&quot;lambda_seq&quot;]]
gg = h5.data[[&quot;genes&quot;]]</code></pre>
<pre class="r"><code>data.dir = &quot;/Users/sabrinami/Library/CloudStorage/Box-Box/rat-genomic-analysis/PTRS-2023/RatData&quot;
traits &lt;- glue::glue(&quot;{data.dir}/phenotypes/obesity_upublished_data_raw_phenotypes_PalmerLab.csv&quot;)
measured.traits &lt;- fread(traits) %&gt;% filter(!is.na(length_wo_tail_cm))
m_height &lt;- measured.traits %&gt;% select(IID=rfid,HGHT=length_wo_tail_cm)</code></pre>
<pre class="r"><code>nn.dta &lt;- measured.traits %&gt;% 
  select(IID=rfid,HGHT=length_wo_tail_cm,SEX=sex)
test &lt;- lm(paste0(&quot;HGHT&quot;, &quot; ~ &quot;, &quot; SEX&quot;),
             data = nn.dta)
adj.height &lt;- data.frame(IID = nn.dta$IID, HGHT = resid(test))</code></pre>
<pre class="r"><code>obs.file &lt;- glue::glue(&quot;{data.dir}/phenotypes/gexBr.csv&quot;)
obs.gene &lt;- fread(obs.file)
obs.gene &lt;- rename(obs.gene, IID=X)
orth.rats = fread(&quot;/Users/sabrinami/Library/CloudStorage/Box-Box/rat-genomic-analysis/PTRS-2023/RatXcan/ortholog_genes_rats_humans.tsv&quot;)

# filter for genes with human ortholog
obs.exp &lt;- obs.gene %&gt;% select(c(IID, intersect(colnames(obs.gene), orth.rats$rnorvegicus_homolog_ensembl_gene) ))



colnames(obs.exp)[2:ncol(obs.exp)] &lt;- orth.rats[match(colnames(obs.exp)[2:ncol(obs.exp)], orth.rats$rnorvegicus_homolog_ensembl_gene), 1] %&gt;% .[[&quot;ensembl_gene_id&quot;]]</code></pre>
<pre class="r"><code>## predicted expression for rats with phenotype data
pred.ctimp = fread(&quot;/Users/sabrinami/Library/CloudStorage/Box-Box/rat-genomic-analysis/PTRS-2023/RatXcan/Results/Br_EN__predict.txt&quot;, header=TRUE)
pred.ctimp$IID = pred.ctimp$FID
pred.ctimp = pred.ctimp %&gt;% select(c(IID, intersect(colnames(pred.ctimp), orth.rats$rnorvegicus_homolog_ensembl_gene) ))

colnames(pred.ctimp)[2:ncol(pred.ctimp)] &lt;- orth.rats[match(colnames(pred.ctimp)[2:ncol(pred.ctimp)], orth.rats$rnorvegicus_homolog_ensembl_gene), 1] %&gt;% .[[&quot;ensembl_gene_id&quot;]]</code></pre>
<div id="festuss-ptrs-functions" class="section level2">
<h2>Festus’s PTRS functions</h2>
<pre class="r"><code>fast_generate_trait &lt;- function(exp, weights){
  # convert the predicted df into a matrix
  test &lt;- exp %&gt;% column_to_rownames(var = &quot;IID&quot;) %&gt;%
    as.matrix()

  weights = weights %&gt;% data.frame() %&gt;%
    column_to_rownames(var = &quot;gene_name&quot;) %&gt;% as.matrix()

  # find the intersect
  ww &lt;- weights[intersect(colnames(test), rownames(weights)), ,drop = FALSE ]
  test &lt;- test[, intersect(colnames(test), rownames(ww))]

  aa = test %*% ww
  return(aa)
}

ext_weights &lt;- function(hd5file) {
  ## get atable of weights for different models from a h5 file
  h5.data &lt;- h5read(hd5file, &quot;/&quot;)
  bb = h5.data[[&quot;dataset_0&quot;]][[&quot;betahat&quot;]]
  ll = h5.data[[&quot;dataset_0&quot;]][[&quot;lambda_seq&quot;]]
  gg = h5.data[[&quot;genes&quot;]]
  
  ptrs.weights &lt;- bb %&gt;% data.frame() %&gt;% 
    setnames(., names(.), gsub(&quot;\\..*&quot;,&quot;&quot;,gg)) %&gt;% 
    rownames_to_column(var = &quot;model&quot;) %&gt;% 
    mutate(model = glue::glue(&quot;model_{model}&quot;)) %&gt;% 
    column_to_rownames(var = &quot;model&quot;) %&gt;% 
    t() %&gt;% data.frame() %&gt;% 
    rownames_to_column(var = &quot;gene_name&quot;)
  
  return(ptrs.weights)
}

run_trait_cor &lt;- function(obs.exp,pred.exp,ptrs.weights,m.trait,adj.height) {
  ## gene.exp: gene expression df IID first col then genes with no version.
  ## ptrs.weights: gene first col then models
  ## m.trait: observed trait IID first col and trait measured second col
  
  pred.cors &lt;- data.frame(model=as.character(), param = as.character(),
                        group = as.character(), corr = as.numeric(),
                        pvalue = as.numeric())
  pred.heights &lt;- m.trait
  
  for (mmm in names(ptrs.weights)[-1]) {
    
    ptrs.temp &lt;- ptrs.weights %&gt;% 
      select(gene_name, all_of(mmm))
    
    ### OBSERVED GENE EXPRESSION
    pred_height &lt;- fast_generate_trait(obs.exp,ptrs.temp) %&gt;% 
      data.frame() %&gt;% setnames(.,names(.), c(&quot;predicted&quot;)) %&gt;% 
      rownames_to_column(var = &quot;IID&quot;)
    
    ## correlations 
    all.cor &lt;- m_height %&gt;% 
      inner_join(pred_height, by = &quot;IID&quot;)
    cor.result &lt;- cor.test(all.cor$HGHT,all.cor$predicted)
    pvalue &lt;- cor.result$p.value
    estimate &lt;- cor.result$estimate
    
    pred.cors &lt;- pred.cors %&gt;% 
      add_row(model = &quot;obs.height&quot;,param = mmm, group = &quot;obs.exp&quot;,
              corr = estimate,pvalue = pvalue)
    
    ## Create a dataframe
    pred.heights &lt;- pred.heights %&gt;% 
      inner_join(pred_height %&gt;% dplyr::rename(!!mmm := predicted)
                 , by = &quot;IID&quot;)
    
    ## Adjusted height
    all.cor &lt;- adj.height %&gt;% 
      inner_join(pred_height, by = &quot;IID&quot;)
    cor.result &lt;- cor.test(all.cor$HGHT,all.cor$predicted)
    pvalue &lt;- cor.result$p.value
    estimate &lt;- cor.result$estimate
    
    pred.cors &lt;- pred.cors %&gt;% 
      add_row(model = &quot;adj.height&quot;, param = mmm, group = &quot;obs.exp&quot;,
              corr = estimate,pvalue = pvalue)
    ######################################
    ##PREDICTED GENE EXPRESSION
    pred_height &lt;- fast_generate_trait(pred.exp,ptrs.temp) %&gt;% 
      data.frame() %&gt;% setnames(.,names(.), c(&quot;predicted&quot;)) %&gt;% 
      rownames_to_column(var = &quot;IID&quot;)
    
    ## correlations 
    all.cor &lt;- m_height %&gt;% 
      inner_join(pred_height, by = &quot;IID&quot;)
    cor.result &lt;- cor.test(all.cor$HGHT,all.cor$predicted)
    pvalue &lt;- cor.result$p.value
    estimate &lt;- cor.result$estimate
    
    pred.cors &lt;- pred.cors %&gt;% 
      add_row(model = &quot;obs.height&quot;,param = mmm, group = &quot;pred.exp&quot;,
              corr = estimate,pvalue = pvalue)
    
    ## Create a dataframe
    pred.heights &lt;- pred.heights %&gt;% 
      inner_join(pred_height %&gt;% dplyr::rename(!!mmm := predicted)
                 , by = &quot;IID&quot;)
    
    ## Adjusted height
    all.cor &lt;- adj.height %&gt;% 
      inner_join(pred_height, by = &quot;IID&quot;)
    cor.result &lt;- cor.test(all.cor$HGHT,all.cor$predicted)
    pvalue &lt;- cor.result$p.value
    estimate &lt;- cor.result$estimate
    pred.cors &lt;- pred.cors %&gt;% 
      add_row(model = &quot;adj.height&quot;, param = mmm, group = &quot;pred.exp&quot;,
              corr = estimate,pvalue = pvalue)
    
  }
  ff &lt;- factor(unique(pred.cors$param))
  plt &lt;- pred.cors %&gt;% 
    mutate(param = factor(param, levels = ff)) %&gt;% 
    select(-pvalue) %&gt;%  spread(group,corr) %&gt;% 
    ggplot( aes(x=param, group = model, color = model)) +
      geom_line(aes(y=obs.exp),linetype=&quot;dotted&quot;) +
      geom_line(aes(y=pred.exp)) +
      theme_classic() +
      theme(axis.text.x=element_blank()) 
  print(plt)
  
  return(list(pred.cors=pred.cors,pred.heights=pred.heights))
}</code></pre>
<pre class="r"><code>print(&quot;trait: length without tail&quot;)</code></pre>
<pre><code>[1] &quot;trait: length without tail&quot;</code></pre>
<pre class="r"><code>org.out = run_trait_cor(obs.exp,pred.ctimp,ext_weights(in.file),m_height,adj.height)</code></pre>
<p><img src="figure/PTRS_Predict.Rmd/plot%20cors-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-cors-1">
Past versions of "plot cors-1.png"
</button>
</p>
<div id="fig-no-spaces-plot-cors-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sabrina-mi/PTRS-Rat-Analysis/blob/791b48ea19824e790d7043c37e2949c951a4fcae/docs/figure/PTRS_Predict.Rmd/plot cors-1.png" target="_blank">791b48e</a>
</td>
<td>
sabrina-mi
</td>
<td>
2023-04-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>org.out$pred.cors %&gt;% filter(group==&quot;obs.exp&quot;) %&gt;% arrange(pvalue)</code></pre>
<pre><code>        model    param   group       corr       pvalue
1  obs.height model_11 obs.exp 0.24645709 0.0001762908
2  obs.height model_10 obs.exp 0.24641363 0.0001767713
3  obs.height  model_5 obs.exp 0.24579154 0.0001837857
4  obs.height  model_1 obs.exp 0.24579151 0.0001837860
5  obs.height  model_2 obs.exp 0.24579151 0.0001837860
6  obs.height  model_3 obs.exp 0.24579151 0.0001837860
7  obs.height  model_4 obs.exp 0.24579151 0.0001837860
8  obs.height  model_6 obs.exp 0.24579095 0.0001837925
9  obs.height  model_7 obs.exp 0.24578944 0.0001838099
10 obs.height  model_8 obs.exp 0.24578687 0.0001838394
11 obs.height  model_9 obs.exp 0.24577632 0.0001839606
12 adj.height model_10 obs.exp 0.09926366 0.1359651084
13 adj.height  model_8 obs.exp 0.09799229 0.1410758435
14 adj.height  model_1 obs.exp 0.09798691 0.1410977472
15 adj.height  model_2 obs.exp 0.09798691 0.1410977472
16 adj.height  model_3 obs.exp 0.09798691 0.1410977472
17 adj.height  model_4 obs.exp 0.09798691 0.1410977472
18 adj.height  model_5 obs.exp 0.09798672 0.1410985301
19 adj.height  model_6 obs.exp 0.09798640 0.1410998468
20 adj.height  model_7 obs.exp 0.09798580 0.1411022685
21 adj.height  model_9 obs.exp 0.09796805 0.1411746778
22 adj.height model_11 obs.exp 0.09544464 0.1517567016</code></pre>
<pre class="r"><code>print(&quot;trait: length with tail&quot;)</code></pre>
<pre><code>[1] &quot;trait: length with tail&quot;</code></pre>
<pre class="r"><code>measured.traits &lt;- fread(traits) %&gt;% filter(!is.na(length_w_tail_cm))
m_height &lt;- measured.traits %&gt;% select(IID=rfid,HGHT=length_w_tail_cm)
nn.dta &lt;- measured.traits %&gt;% 
  select(IID=rfid,HGHT=length_w_tail_cm,SEX=sex)
test &lt;- lm(paste0(&quot;HGHT&quot;, &quot; ~ &quot;, &quot; SEX&quot;),
             data = nn.dta)
adj.height &lt;- data.frame(IID = nn.dta$IID, HGHT = resid(test))

org.out = run_trait_cor(obs.exp,pred.ctimp,ext_weights(in.file),m_height,adj.height)</code></pre>
<p><img src="figure/PTRS_Predict.Rmd/length%20w%20tail%20corrs-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-length-w-tail-corrs-1">
Past versions of "length w tail corrs-1.png"
</button>
</p>
<div id="fig-no-spaces-length-w-tail-corrs-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sabrina-mi/PTRS-Rat-Analysis/blob/791b48ea19824e790d7043c37e2949c951a4fcae/docs/figure/PTRS_Predict.Rmd/length w tail corrs-1.png" target="_blank">791b48e</a>
</td>
<td>
sabrina-mi
</td>
<td>
2023-04-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>org.out$pred.cors %&gt;% filter(group==&quot;obs.exp&quot;) %&gt;% arrange(pvalue)</code></pre>
<pre><code>        model    param   group      corr       pvalue
1  obs.height model_11 obs.exp 0.2454144 0.0001881643
2  obs.height model_10 obs.exp 0.2436298 0.0002102324
3  obs.height  model_5 obs.exp 0.2430405 0.0002180354
4  obs.height  model_1 obs.exp 0.2430405 0.0002180360
5  obs.height  model_2 obs.exp 0.2430405 0.0002180360
6  obs.height  model_3 obs.exp 0.2430405 0.0002180360
7  obs.height  model_4 obs.exp 0.2430405 0.0002180360
8  obs.height  model_6 obs.exp 0.2430399 0.0002180438
9  obs.height  model_7 obs.exp 0.2430380 0.0002180691
10 obs.height  model_8 obs.exp 0.2430242 0.0002182553
11 obs.height  model_9 obs.exp 0.2430141 0.0002183917
12 adj.height model_10 obs.exp 0.1083706 0.1034076488
13 adj.height model_11 obs.exp 0.1071747 0.1072937555
14 adj.height  model_1 obs.exp 0.1069974 0.1078795801
15 adj.height  model_2 obs.exp 0.1069974 0.1078795801
16 adj.height  model_3 obs.exp 0.1069974 0.1078795801
17 adj.height  model_4 obs.exp 0.1069974 0.1078795801
18 adj.height  model_5 obs.exp 0.1069972 0.1078802058
19 adj.height  model_6 obs.exp 0.1069968 0.1078815624
20 adj.height  model_7 obs.exp 0.1069955 0.1078859584
21 adj.height  model_8 obs.exp 0.1069829 0.1079276888
22 adj.height  model_9 obs.exp 0.1069565 0.1080151270</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 (2022-10-31)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] rhdf5_2.42.0      glmnet_4.1-6      Matrix_1.5-3      RSQLite_2.3.0    
 [5] data.table_1.14.8 lubridate_1.9.2   forcats_1.0.0     stringr_1.5.0    
 [9] dplyr_1.1.1       purrr_1.0.1       readr_2.1.4       tidyr_1.3.0      
[13] tibble_3.2.1      ggplot2_3.4.1     tidyverse_2.0.0   workflowr_1.7.0  

loaded via a namespace (and not attached):
 [1] fs_1.6.1            usethis_2.1.6       devtools_2.4.5     
 [4] bit64_4.0.5         httr_1.4.5          rprojroot_2.0.3    
 [7] gh_1.4.0            tools_4.2.2         profvis_0.3.7      
[10] bslib_0.4.2         utf8_1.2.3          R6_2.5.1           
[13] DBI_1.1.3           colorspace_2.1-0    rhdf5filters_1.10.0
[16] urlchecker_1.0.1    withr_2.5.0         tidyselect_1.2.0   
[19] prettyunits_1.1.1   processx_3.8.0      curl_5.0.0         
[22] bit_4.0.5           compiler_4.2.2      git2r_0.31.0       
[25] httr2_0.2.2         cli_3.6.0           labeling_0.4.2     
[28] sass_0.4.5          scales_1.2.1        callr_3.7.3        
[31] rappdirs_0.3.3      digest_0.6.31       rmarkdown_2.20     
[34] pkgconfig_2.0.3     htmltools_0.5.4     sessioninfo_1.2.2  
[37] highr_0.10          fastmap_1.1.1       htmlwidgets_1.6.2  
[40] rlang_1.1.0         rstudioapi_0.14     shiny_1.7.4        
[43] farver_2.1.1        shape_1.4.6         jquerylib_0.1.4    
[46] generics_0.1.3      jsonlite_1.8.4      magrittr_2.0.3     
[49] Rhdf5lib_1.20.0     Rcpp_1.0.10         munsell_0.5.0      
[52] fansi_1.0.4         lifecycle_1.0.3     stringi_1.7.12     
[55] whisker_0.4.1       yaml_2.3.7          pkgbuild_1.4.0     
[58] grid_4.2.2          blob_1.2.4          promises_1.2.0.1   
[61] crayon_1.5.2        miniUI_0.1.1.1      lattice_0.20-45    
[64] splines_4.2.2       hms_1.1.3           knitr_1.42         
[67] ps_1.7.3            pillar_1.9.0        codetools_0.2-19   
[70] pkgload_1.3.2       glue_1.6.2          evaluate_0.20      
[73] getPass_0.2-2       remotes_2.4.2       vctrs_0.6.1        
[76] tzdb_0.3.0          httpuv_1.6.9        foreach_1.5.2      
[79] gtable_0.3.3        cachem_1.0.7        xfun_0.37          
[82] mime_0.12           xtable_1.8-4        gitcreds_0.1.2     
[85] later_1.3.0         survival_3.5-5      iterators_1.0.14   
[88] memoise_2.0.1       timechange_0.2.0    ellipsis_0.3.2     </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
